ui/src/router/components/form/components/experiment_config/validation/schema.js

```diff
@@ -15,21 +15,17 @@ const experimentSchema = yup.object().shape({
 
 const variableSchema = yup.object().shape({
   required: yup.boolean(),
-  field_source: yup.mixed().when("required", {
+  field_source: yup.string().when("required", {
     is: true,
-    then: fieldSourceSchema.required(
-      "Select the field source for the required variable"
-    ),
+    then: fieldSourceSchema,
   }),
-  field: yup.mixed().when("required", {
+  field: yup.string().when("required", {
     is: true,
-    then: fieldSchema("field_source").required(
-      "Specify the field name for the required variable"
-    ),
+    then: fieldSchema("field_source"),
```

Comment:
These `required` conditions were removed because we can no longer call `required` on a function (`fieldSchema("field_source")` is a function). This isn't a problem because the schema returned by `fieldSchema("field_source")` itself contains very similar error prompts: